src/boot/bootctl-install.c

```diff
@@ -489,9 +644,13 @@ static int install_binaries(const char *esp_path, const char *arch, bool force)
                  * newer version, or other boot loader in place. */
                 if (arg_graceful && IN_SET(k, -ESTALE, -ESRCH))
                         continue;
+
                 RET_GATHER(r, k);
         }
 
+        if (r >= 0 && ret_default_loader_path)
+                *ret_default_loader_path = strjoin("/EFI/systemd/systemd-boot", arch, ".efi");
```

Comment:
oom check missing. i.e. pleas assign strjoin() return path to local variable first, and only on success move to `*ret_default_loader_path`.